packages/server-wallet/docs/index.md

```diff
@@ -0,0 +1,183 @@
+
+```
+                                                    |
+                     +-----------------+            |     +--------------+
+                     |                 |            |     |              |
+                     |  CHAIN-SERVICE  <------------------>  BLOCKCHAIN  |
+                     |                 |            |     |              |
+                     +--------^--------+            |     +--------------+
+                              |                     |
+                              |                     |
+                        +-----v----+                |     +----------+
+                        |          |                |     |          |
+                        |          |                |     |          |
+                        |          |  API Request   |     |          |
+                        |          <-----------------------          |
+                        |          |  API Response  |     |          |
+         +---------+    |          ----------------------->          |
+         |         |    |          |                |     |          |
+         |  STORE  <--->|  WALLET  |  Events        |     |   APP    |
+         |         |    |          ----------------------->          |
+         +---------+    |          |                |     |          |
+                        |          |                |     |          |
+                        |          |  pushMessage   |     |          |
+                        |          <---------------------------+     |
+                        |          |                |     |    |     |
+                        |          |                |     |    |     |
+                        +----------+                |     +----|-----+
+                                                    |          |
+                                                    | +--------|---------+
+                                                    | |  COUNTERPARTY    |
+                                                    | |  WALLETS         |
+                                                      +------------------+
+```
+
+# The responsibilities of a state channel wallet
+
+As in the [Hippocratic oath](https://en.wikipedia.org/wiki/Hippocratic_Oath): _primum non nocere,_ (first, do no harm).
+
+1. Do not lose secrets (private keys)
+2. Do not leak secrets (private keys)
+3. Store relevant states and signatures (Don’t lose any unless they are screened off by kept ones, allow states to be queried)
+4. Do not sign states or blockchain transactions unless the App grants permission
+
+Next, do some active things to protect The App’s interests.
+
+5. With implicit permission, allow an exit on chain without losing funds
+6. Detect and store blockchain challenges (for querying)
+7. Detect and store changes in channel funding (for querying)
```

Comment:
It might even be more clear not to tell the audience that we store these things. I think this is the _intent_ of the wallet, from an "external point of view".

```suggestion
6. React to blockchain challenges
7. React to changes in channel funding
```

Reply:
In my mind, reacting to these events is a separate responsibility of the wallet that builds on the more basic responsibility of storing those events. If a partially-functional wallet only stores the challenges but does not otherwise react to them, or if the code that reacts to challenge is somehow buggy, at least users can attempt to recover their funds manually by falling back on the more basic responsibility.

Reply:
Could one equivalently argue that a wallet which stores states but does not sign new states (for an objective) exhibits the "basic responsibility" of being a wallet? In that case, a user can attempt to fund their app channel by signing ledger updates.

It certainly feels as though the wallet has an _obligation_ to react to a `Challenge` event in some fashion. It may react by calling `checkpoint`, if it detects that the challenge was malicious. It may also react by notifying The App:  "⚠️ !!! A challenge was detected! ⚠️".

Reply:
Yes I agree, a fully featured wallet should react to challenges. I suppose the direction I am trying to go in, with this ordered list of responsibilities, is that they are built up in a certain order based on "security priority". In that case, reacting to a challenge would be a separately-enumerated responsibility with a lower priority. A wallet should still meet that responsibility in order to be considered feature-complete; but it should not meet it without first meeting all other responsibilities with a higher priority. 

By placing emphasis on _storage_ high up the priority list, I suppose I am trying to discourage wallet builders from implementing features that do not survive a crash/restart cycle.  It's a "walk before you run" approach.